packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm

```diff
@@ -602,6 +607,19 @@ - (void)scrollViewWillEndDragging:(UIScrollView *)scrollView
       targetContentOffset->y = scrollView.contentOffset.y + travel * _endDraggingSensitivityMultiplier;
     }
   }
+
+  if (!_eventEmitter) {
+    return;
+  }
+
+  auto metrics = ScrollViewEventEmitter::EndDragMetrics{};
+  [self _setScrollViewMetrics:metrics];
+  metrics.targetContentOffset.x = targetContentOffset->x;
+  metrics.targetContentOffset.y = targetContentOffset->y;
+  metrics.velocity.x = velocity.x;
+  metrics.velocity.y = velocity.y;
+
+  static_cast<const ScrollViewEventEmitter &>(*_eventEmitter).onScrollEndDrag(metrics);
```

Comment:
This changes the semantic of when the event is going to be fired.
Before it was firing when the scrollview stopped scrolling (DidEnd). Now it fires while it is still scrolling (WillEnd). This change in behavior is a breaking change and can introduce subtle bugs.

Can we adopt a different approach, where we store the `targetContentOffset` and the `velocity` locally, and then fire the event in the `didEnd`, using the value we stored?

We should also clear the values after the event is fired.

Reply:
@cipolleschi Actually the paper just fires it in the `willEnd`.
https://github.com/facebook/react-native/blob/b04d17afcac82af1a47fd462a04fe4088d19b468/packages/react-native/React/Views/ScrollView/RCTScrollView.m#L859

Reply:
My bad then, I have not checked with the old arch. Let's keep it in the willEnd, then.